fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckTelemetryMiddlewareTest.kt

```diff
@@ -234,4 +247,123 @@ class ReviewQualityCheckTelemetryMiddlewareTest {
 
         assertNotNull(Shopping.surfacePoweredByFakespotLinkClicked.testGetValue())
     }
+
+    @Test
+    fun `GIVEN a product review has been updated WHEN stale analysis is present and product is not in analysis THEN the stale analysis event is recorded`() {
+        val engineSession = mockk<EngineSession>()
+        val productTab = createTab(
+            url = "pdp",
+            id = "1",
+            isProductUrl = true,
+            engineSession = engineSession,
+        )
+        val browserState = BrowserState(
+            tabs = listOf(productTab),
+            selectedTabId = productTab.id,
+        )
+        val testedStore = ReviewQualityCheckStore(
+            initialState = ReviewQualityCheckState.OptedIn(
+                productReviewState = ProductAnalysisTestData.analysisPresent(
+                    analysisStatus = AnalysisPresent.AnalysisStatus.UP_TO_DATE,
+                ),
+                productRecommendationsPreference = false,
+                productVendor = ReviewQualityCheckState.ProductVendor.AMAZON,
+            ),
+            middleware = listOf(
+                ReviewQualityCheckTelemetryMiddleware(BrowserStore(browserState), appStore),
+            ),
+        )
+        val testedReviewState = ReviewQualityCheckState.OptedIn(
+            productReviewState = ProductAnalysisTestData.analysisPresent(
+                analysisStatus = AnalysisPresent.AnalysisStatus.NEEDS_ANALYSIS,
+            ),
+            productRecommendationsPreference = false,
+            productVendor = ReviewQualityCheckState.ProductVendor.AMAZON,
+        )
+
+        testedStore.dispatch(ReviewQualityCheckAction.UpdateProductReview(testedReviewState.productReviewState)).joinBlocking()
+        testedStore.waitUntilIdle()
+
+        assertNotNull(Shopping.surfaceStaleAnalysisShown.testGetValue())
+    }
+
+    @Test
+    fun `GIVEN a product review has been updated WHEN stale analysis is present and product is being analyzed THEN the stale analysis event is not recorded`() {
+        val engineSession = mockk<EngineSession>()
+        val productTab = createTab(
+            url = "pdp",
+            id = "1",
+            isProductUrl = true,
+            engineSession = engineSession,
+        )
+        appStore.dispatch(AppAction.ShoppingAction.AddToProductAnalysed("pdp")).joinBlocking()
+        appStore.waitUntilIdle()
+        val browserState = BrowserState(
+            tabs = listOf(productTab),
+            selectedTabId = productTab.id,
+        )
+        val testedStore = ReviewQualityCheckStore(
+            initialState = ReviewQualityCheckState.OptedIn(
+                productReviewState = ProductAnalysisTestData.analysisPresent(
+                    analysisStatus = AnalysisPresent.AnalysisStatus.NEEDS_ANALYSIS,
```

Comment:
Not sure we need this since it's the `AppStore` state condition that is being tested here.